File: Easy/Equilibrium Point/equilibrium_point.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;
import java.util.stream.*;

class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t =
            Integer.parseInt(br.readLine().trim()); // Inputting the testcases
        while (t-- > 0) {
            
            //taking input n
            int n = Integer.parseInt(br.readLine().trim());
            long arr[] = new long[n];
            String inputLine[] = br.readLine().trim().split(" ");
            
            //adding elements to the array
            for (int i = 0; i < n; i++) {
                arr[i] = Long.parseLong(inputLine[i]);
            }

            Solution ob = new Solution();
            
            //calling equilibriumPoint() function
            System.out.println(ob.equilibriumPoint(arr, n));
        }
    }
}
// } Driver Code Ends


class Solution {

    
    // a: input array
    // n: size of array
    // Function to find equilibrium point in the array.
    public static int equilibriumPoint(long arr[], int n) {

        // Your code here
        long total = 0;
        for(int i = 0; i<n; i++){
            total = total + arr[i];
        }
        
        long left_sum = 0;
        long right_sum = 0;
        
        for(int i=0; i < n; i++){
            
            right_sum = total - left_sum - arr[i];
            
            
            if (left_sum == right_sum){
                return i+1;
            }
            else{
                left_sum = left_sum + arr[i];
            }
        }
        return -1;
    }
}

File: Easy/Equilibrium Point/equilibrium_point.py
class Solution:
    # Complete this function
    
    #Function to find equilibrium point in the array.
    def equilibriumPoint(self,A, N):
        
        total = 0
        
        # To find the total sum of given array
        for i in A:
            total = total + i
            
        left_sum = 0
        # right_sum = 0
        
        for i in range(0, len(A)):
            right_sum = total - left_sum - A[i]
            
            if (left_sum == right_sum):
                return i+1
            else:
                left_sum = left_sum + A[i]
        
        return -1

